src/release/breaking-changes/flutter-gradle-plugin-apply.md

```diff
@@ -0,0 +1,195 @@
+---
+title: Deprecated imperative apply of Flutter Gradle Plugin
+description: >
+  How to migrate your Android app's Gradle build files to the new, declarative
+  format.
+---
+
+## Summary
+
+To build a Flutter app for Android, the Flutter Gradle plugin must be applied.
+Historically, this was done imperatively with Gradle's
+[legacy, imperative apply method][].
+
+In Flutter 3.16, support has been added for applying these plugins with Gradle's
+[declarative plugins {} block][], and it is the recommended approach. Since
+Flutter 3.16, projects generated with `flutter create` use the new apply method
+of applying Gradle plugins. Older projects, however, should be migrated
```

Comment:
thanks, I mostly agree, done in 13507699bb09b0438555f56223e8a2564def76f0. I went with:

> Projects created with versions of Flutter prior to 3.16 need to be migrated manually.

because I'm not aware of any work happening on `flutter migrate`, which is the only potential solution to automate this migration.

Also I understand your good intentions with this:

> the build migration (link to steps) should take one engineer less than a day.

but we don't have data to back that up (except personal experience). I've seen big Flutter projects that have a lot of custom Gradle conifg and I just wouldn't be comfortable with saying this (haha, I'm referring to personal experience now, but you get me). I think it might sometimes take more than 1 day, esp. if the engineer is very new to Gradle. I'd not include it.

I included this:

> Applying Gradle plugins using the `plugins {}` executes the same code as before
and should produce equivalent app binaries.

in 3308f5aaee0b2a4bdb3ce62a73627721ce53aedd